### ISSUE TITLE
Mutant drops frozen string literal comment for unmutated code

### ISSUE THREAD

**Issue by krowpu** [mbj/mutant#743]
I have a file which contains the following code:

```ruby
# frozen_string_literal: true

module Utils
  class CommonSize
    attr_reader :width, :height

    def initialize(width, height)
      # ...
    end

    def to_s
      @to_s ||= "#{width}x#{height}"
    end
  end
end
```

I have a test for this code:

```ruby
RSpec.describe Utils::CommonSize do
  subject { described_class.new width, height }

  let(:width)  { rand 1..900 }
  let(:height) { rand 1..900 }

  describe '#to_s' do
    specify do
      expect(subject.to_s).to be_frozen
    end
  end
end
```

Mutant fails with the following log:

```
-----------------------
Utils::CommonSize#to_s:/builds/gems/utils/lib/utils/common_size.rb:56
- rspec:12:./spec/lib/utils/common_size_spec.rb:98/Utils::CommonSize#to_s 
- rspec:13:./spec/lib/utils/common_size_spec.rb:102/Utils::CommonSize#to_s 
- rspec:14:./spec/lib/utils/common_size_spec.rb:106/Utils::CommonSize#to_s 
- rspec:15:./spec/lib/utils/common_size_spec.rb:110/Utils::CommonSize#to_s 
neutral:Utils::CommonSize#to_s:/builds/gems/utils/lib/utils/common_size.rb:56:b24e3
--- Neutral failure ---
Original code was inserted unmutated. And the test did NOT PASS.
Your tests do not pass initially or you found a bug in mutant / unparser.
Subject AST:
s(:def, :to_s,
  s(:args),
  s(:or_asgn,
    s(:ivasgn, :@to_s),
    s(:dstr,
      s(:begin,
        s(:send, nil, :width)),
      s(:str, "x"),
      s(:begin,
        s(:send, nil, :height)))))
Unparsed Source:
def to_s
  @to_s ||= "#{width}#{"x"}#{height}"
end
Test Result:
- 4 @ runtime: 0.055803942
  - rspec:12:./spec/lib/utils/common_size_spec.rb:98/Utils::CommonSize#to_s 
  - rspec:13:./spec/lib/utils/common_size_spec.rb:102/Utils::CommonSize#to_s 
  - rspec:14:./spec/lib/utils/common_size_spec.rb:106/Utils::CommonSize#to_s 
  - rspec:15:./spec/lib/utils/common_size_spec.rb:110/Utils::CommonSize#to_s 
Test Output:

Randomized with seed 16072
...F

Failures:

  1) Utils::CommonSize#to_s should be frozen
     Failure/Error: expect(subject.to_s).to be_frozen
       expected `"900x264".frozen?` to return true, got false
     # ./spec/lib/utils/common_size_spec.rb:103:in `block (3 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-rspec-0.8.15/lib/mutant/integration/rspec.rb:63:in `call'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/env.rb:48:in `block in run_mutation_tests'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:66:in `block in result'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant.rb:223:in `open'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant.rb:223:in `block in <class:Config>'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:63:in `result'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:55:in `child'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:37:in `block in parent'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:36:in `fork'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:36:in `parent'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:23:in `block in call'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:22:in `pipe'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/isolation/fork.rb:22:in `call'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/env.rb:46:in `run_mutation_tests'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/env.rb:25:in `kill'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/worker.rb:60:in `call'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/worker.rb:60:in `handle_job'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/worker.rb:45:in `handle'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/worker.rb:31:in `run'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/worker.rb:17:in `run'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/parallel/master.rb:43:in `block (2 levels) in run'
     # ./vendor/bundle/ruby/2.3.0/gems/mutant-0.8.15/lib/mutant/actor/env.rb:14:in `block in spawn'

Finished in 0.02702 seconds (files took 45.09 seconds to load)
4 examples, 1 failure

Failed examples:

rspec ./spec/lib/utils/common_size_spec.rb:102 # Utils::CommonSize#to_s should be frozen

Randomized with seed 16072

-----------------------
```

**Comment by mbj** [comment]
@krowpu Mutant needs to be made aware of the magic comment to reproduce it on insertion of mutations.

I doubt I've time to do it soon, until this specific bug is in the blocking path of my commercial activities. PRs welcome.

**Comment by mbj** [comment]
This is part way (updating to a parser that has more infrastructure for this): https://github.com/mbj/mutant/pull/746

**Comment by mbj** [comment]
@krowpu This was addressed via #818
